### DISCUSSION TITLE
Annotazioni automatiche - Allineamento db locale con ANSC

### DISCUSSION THREAD

**Discussion by Abstractconstruct** [italia/ansc#735] [Supporto servizi cooperativi] [answered]
Salve,

Per quanto riguarda il flusso di conferma delle notifiche di atti digitali presenti in ANSC, per effettuare l'allineamento locale è necessario in seguito alla ricerca R008, la consultazione R005 dell'annotazione automatica. 

Il dubbio che ha bisogno di chiarimenti è: per effettuare l'inserimento dell'annotazione il nostro applicativo ha bisogno di un modo per identificare in locale l'atto primario su cui apporre l'annotazione. 
Non abbiamo potuto trovare un identificativo come id ANSC (che memorizziamo localmente), oppure il numero atto comunale, tra i json di risposta dei servizi cooperativi r008 ricerca e r005 consultazione. Tra i dati ci sono solamente l'identificativo dell'atto scatenante (nome della proprietà: eventoCollegato del modelEvento) e l'id del soggetto intestatario.  

Per cui la domanda principale è se viene previsto da qualche servizio recuperare un riferimento dell'atto digitale primario su cui verrà inserito l'annotazione dopo la conferma alla proposta?

Distinti Saluti.
Rimaniamo in attesa di vostra indicazione in merito

**Comment by Abstractconstruct** [comment]
Buongiorno, ci sono aggiornamenti?

**Comment by gbuemi** [comment]
Buonasera @Abstractconstruct,
vi invitiamo a consultare la [Nota processo Flusso di Conferma](https://github.com/italia/ansc/blob/main/docs/Note/Nota_Processo_Flusso_Conferma/Nota_Processo_Flusso_Conferma.pdf) pubblicata nella sezione docs del repository

Un saluto,
Supporto ANSC

**Reply by Abstractconstruct** [reply]
Salve,

avevo già visionato il link, e l'ho riletta. Viene descritto il flusso di conferma.  Sulla sezione proposta di annotazione automatica purtroppo non viene descritto/oppure non riesco a trovare come recuperare l'atto primario su cui il sistema ANSC apporrà l'annotazione automatica. 

I servizi citati nelle varie documentazioni sulle note sono ricerca r008 delle notifiche e consultazione R005 dell'annotazione. Di cui payload di risposta non contengono identificativi dell'atto primario.

Es.  Sono dati di soggetti test in preprod
ModelNotifica da R008:
  {
    "idSoggettoRif": "2261277",
    "descrizione": "Annotazione automatica id 2024-1219857-00000-052032",
    "idStatoFlussoNotifiche": "1",
    "idGenereNotifiche": "1",
    "idAnsc": "2024-1219857-00000-052032",
    "idComuneCompetente": "9493",
    "idcanalenotifica": 0,
    "dataInizioValidita": "2024-02-06T09:14:31Z",
    "id": "2261492",
    "dataFineValidita": "9999-12-31T09:14:31Z",
    "idComuneFormazione": "9493"
  }

ModelEvento dell'annotazione da R005:
  {
    "datains": "2024-02-06T09:14:32.463635",
    "operatoreNome": "INF.OR.",
    "idTipoEvento": "1",
    "minuto": "14",
    "dataultimoagg": "2024-02-06T09:14:32.463635",
    "composizioneCompleta": "EREMITA PAPINI è morto in SIENA (SI) il 06/02/2024. Atto di morte registrato col numero 2024-1219856-667-052032",
    "operatoreCognome": "INF.OR.",
    "idVersion": 100008,
    "idComuneRegistrazione": "9493",
    "idUsecase": "10000000",
    "idProvinciaRegistrazione": "052",
    "stato": "GENERATO DA SISTEMA",
    "idAnsc": "2024-1219857-00000-052032",
    "siglaProvinciaRegistrazione": "SI",
    "idtipocontenuto": "4",
    "redattoNellaCasaComunale": true,
    "id": 1219857,
    "operatore": "COMUNETEST052032",
    "eventoCollegato": {
      "idProvinciaRegistrazione": "052",
      "idAnsc": "2024-1219856-667-052032",
      "siglaProvinciaRegistrazione": "SI",
      "numeroatto": "667",
      "idtipocontenuto": "1",
      "idComuneRegistrazione": "9493",
      "annoAtto": "2024",
      "dataformazione": "2024-02-06",
      "nomeComuneRegistrazione": "SIENA",
      "tipologia": "2"
    },
    "dataformazione": "2024-02-06",
    "ruoloOperatore": "2",
    "ora": "10",
    "nomeComuneRegistrazione": "SIENA"
  }

E' necessario per allinearsi con ANSC, recuperare su quale atto verrà aggiunta l'annotazione automatica. 
Visto che non viene descritto, la domanda è se l'inserimento dell'annotazione automatica nel nostro applicativo deve essere effettuato manualmente dall'operatore ricercando sulla webapp per esempio, oppure alla vecchia maniera, altrimenti tramite identificativo ANSC dell'atto primario pervenuto dai servizi cooperativi di ANSC.

Distinti Saluti,
Rimaniamo in attesa di risposta.

**Reply by Abstractconstruct** [reply]
@gbuemi

**Comment by gbuemi** [comment, marked as answer]
@Abstractconstruct dai servizi attualmente fruibili non è possibile ricavare l'atto primario su cui il sistema ANSC apporrà l'annotazione automatica.

Un saluto,
Supporto ANSC

**Comment by ricpapini** [comment]
Salve, non essendo possibile ricavare l'atto primario su cui verrà apportata l'annotazione automatica, come possiamo fare registrala in tale atto nella banca dati locale?

**Comment by gbuemi** [comment]
Buongiorno @ricpapini, il processo di apposizione dell'annotazione automatica è gestito interamente dall'infrastruttura ANSC e, per una completa integrazione col sistema, non è necessario un servizio per ricavare l'atto primario su cui il sistema ANSC la apporrà.

Le SH attualmente in adozione controllata non hanno avuto questa esigenza.

Rimaniamo a disposizione per eventuali chiarimenti in merito.

Un saluto,
Supporto ANSC
